authentication/views.py

```diff
@@ -306,27 +306,27 @@ def dispatch(self, request, *args, **kwargs):
     def validate_and_return_user(request, access_token):
         try:
             token = access_token
+            headers = {
+                "Authorization": f"Bearer {token}",
+                "Accept": "application/vnd.github.v3+json",
+            }
+            response = requests.get("https://api.github.com/user", headers=headers)
+            user_data = response.json()
+            user_name = user_data.get("name")
+            user_email = user_data.get("email")
+            try:
+                return User.objects.get(email=user_email)
+            except User.DoesNotExist:
+                logging.info("[Github Oauth] User does not exist. Creating new user.")
+                return User.objects.create_user(
+                    email=user_email, username=user_name, password=None
+                )
         except (
             OAuthError,
             OAuth2Error,
```

Comment:
again, we can't just copy/paste and leave the code like that is. Why do we need to keep using this exceptions handling for this case?